common/views/components/InputTypeNumber/InputTypeNumber.js

```diff
@@ -0,0 +1,44 @@
+// @flow
+import { forwardRef, type ComponentType } from 'react';
+import styled from 'styled-components';
+import Space, { type SpaceComponentProps } from '../styled/Space';
+import { classNames } from '../../../utils/classnames';
+
+const StyledInput: ComponentType<SpaceComponentProps> = styled(Space).attrs({
+  className: classNames({}),
+})`
+  outline: none;
+  border: 1px solid ${props => props.theme.colors.pumice};
+  padding: 12px;
+
+  /* removes up and down arrows webkit adds to number inputs on desktop */
+  &::-webkit-inner-spin-button,
+  &::-webkit-outer-spin-button {
+    -webkit-appearance: none;
+    margin: 0;
+  }
+
+  &:focus {
+    border: 2px solid ${props => props.theme.colors.yellow};
+    padding: 11px;
+  }
+`;
+
+type Props = {
+  label: string,
+  // We can also pass inputProps here
+};
+
+// $FlowFixMe (forwardRef)
```

Comment:
This is supported now https://medium.com/flow-type/supporting-react-forwardref-and-beyond-f8dd88f35544

It'd be good to see that we've exhausted type checking here, as it seems complicated, which is a good reason to have it so we know how to use this component.